Source/Core/AudioCommon/Mixer.cpp

```diff
@@ -282,9 +292,14 @@ void Mixer::PushStreamingSamples(const short* samples, unsigned int num_samples)
 void Mixer::PushWiimoteSpeakerSamples(const short* samples, unsigned int num_samples,
                                       unsigned int sample_rate_divisor)
 {
-  short samples_stereo[MAX_SAMPLES * 2];
-
-  if (num_samples < MAX_SAMPLES)
+  // Max 20 bytes/speaker report, may be 4-bit ADPCM so multiply by 2
+  static constexpr u32 MAX_SPEAKER_SAMPLES = 20 * 2;
+  std::array<short, MAX_SPEAKER_SAMPLES * 2> samples_stereo;
+
+  ASSERT_MSG(AUDIO, num_samples < MAX_SPEAKER_SAMPLES,
```

Comment:
This needs to be `num_samples <= MAX_SPEAKER_SAMPLES`... Actually I think this applies to all of the asserts as well as the `if` a few lines down.